### PR TITLE
Add diamond punch dagger

### PR DESCRIPTION
**Pull request by mightyagrippa**
A late game luxury option for devoted martial artists. Follows the lead of other diamond weapons for stat alterations (+25% total damage, +50% value, +5 butcher, minus a little weight) and uses a slightly changed diamond knife recipe for charcoal and hydrogen amounts.

### CODE REVIEW

File: data/json/recipes/recipe_weapon.json [hunk 1]
```diff
@@ -4022,6 +4022,44 @@
...
+  "category": "CC_WEAPON",
+  "subcategory": "CSC_WEAPON_PIERCING",
+  "skill_used": "fabrication",
+  "difficulty": 10,
+  "time": 220000,
+  "reversible": false,
+  "autolearn": false,
+  "book_learn": [[ "recipe_lab_cvd", 9 ]],
+  "qualities":[
+    {"id":"HAMMER","level":3,"amount":1},
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_weapon.json]: `"amount":1` is redundant.

File: data/json/recipes/recipe_weapon.json [hunk 2]
```diff
@@ -4022,6 +4022,44 @@
...
 },{
   "type" : "recipe",
+  "result": "diamond_punch_dagger",
+  "category": "CC_WEAPON",
+  "subcategory": "CSC_WEAPON_PIERCING",
+  "skill_used": "fabrication",
+  "difficulty": 10,
+  "time": 220000,
+  "reversible": false,
+  "autolearn": false,
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_weapon.json]: `"reversible": false` and `"autolearn": false` are redundant.

File: data/json/recipes/recipe_weapon.json [hunk 3]
```diff
@@ -4022,6 +4022,44 @@
...
+  "book_learn": [[ "recipe_lab_cvd", 9 ]],
+  "qualities":[
+    {"id":"HAMMER","level":3,"amount":1},
+    { "id": "CHISEL", "level": 3 }
+  ], "tools": [
+      [
+      ["cvd_machine", -1]
+      ],
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_weapon.json]: Should be on one line, like:
```
[ ["cvd_machine", -1] ],
```

File: data/json/recipes/recipe_weapon.json [hunk 4]
```diff
@@ -4022,6 +4022,44 @@
...
+      [
+      [ "oxy_torch", 2 ],
+      [ "welder", 10 ],
+      [ "welder_crude", 15 ],
+      [ "toolset", 15 ]
+      ]
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_weapon.json]: This should either get an extra indent or be all on one line.

**Comment by nikheizen** [comment]
Supernatural timing with this PR! :laughing: :

**Comment by mightyagrippa** [comment]
Why are "reversible": false and "autolearn": false redundant? Those entries seem universal across recipe_weapon regardless of true or false.

**Comment by Coolthulhu** [comment]
They're the default. There is no need to state default fields everywhere.
It was needed earlier, but isn't anymore. Maybe later on we'll actually warn on redundant fields on load.

**Comment by Coolthulhu** [comment]
Now that no diamond weapon mod is in, this should also be added to the mod.

**Comment by codemime** [comment]
> Maybe later on we'll actually warn on redundant fields on load.

I like the idea a lot and it's fairly easy to implement.

**Comment by mightyagrippa** [comment]
Added to blacklist. Extra commits are me reaping the whirlwind for editing in both web browser and notepad+

**Comment by mugling** [comment]
> I like the idea a lot and it's fairly easy to implement.

Incorrectly spelled fields are trivial to detect via the lint tool. Inheritance makes redundant fields a little more tricky (but not impossible). The only way I can think of making this work is for `assign` to warn if a call to it was effectively a no-op. If you have other suggestions I'd definitely be interested.

File: data/json/items/melee.json [hunk 5]
```diff
@@ -1037,6 +1037,23 @@
     },
     {
         "type":"GENERIC",
+        "id" : "diamond_punch_dagger",
```
Comment (mugling) [inline data/json/items/melee.json]: Ideal candidate for `copy-from`

**Pull request merged**